python/tests/test_dataset.py

```diff
@@ -8,9 +8,23 @@
 import scipp as sc
 
 
+def test_shape():
+    a = sc.Variable(value=1)
+    d = sc.Dataset(data={'a': a})
+    assert d.shape == []
+    a = sc.Variable(['x'], shape=[2])
+    b = sc.Variable(['y', 'z'], shape=[3, 4])
+    d = sc.Dataset(data={'a': a, 'b': b})
+    assert d.shape == [4, 3, 2]
```

Comment:
Are we making any promises about dimension order in `Dataset`? In principle it is undefined.